blackbox_test/basic_usage.ts

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable */
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import util from 'util';
+import Web3 from 'web3';
+// import WebSocketProvider from 'web3-providers-ws';
+import ERC20Token from './fixtures/build/ERC20Token.json';
+
+const isWs = (backendMode: string) => backendMode === 'ws';
+
+const maxNumberOfAttempts = 10;
+const intervalTime = 5000; // ms
+
+const waitForOpenConnection = async (
+	web3: Web3,
+	backenMode: string,
+	currentAttempt = 1,
+	status = 'connected',
+) =>
+	new Promise<void>((resolve, reject) => {
+		if (!isWs(backenMode)) {
+			resolve();
+			return;
+		}
+
+		const interval = setInterval(() => {
+			if (currentAttempt > maxNumberOfAttempts - 1) {
+				clearInterval(interval);
+				reject(new Error('Maximum number of attempts exceeded'));
+				//not type safe, but, npm installing other packages creates an error
+				//see here: https://github.com/ChainSafe/web3.js/runs/7545108665?check_suite_focus=true
+			} else if ((web3.provider as any).getStatus() === status) {
+				clearInterval(interval);
+				resolve();
+			}
+			// eslint-disable-next-line no-plusplus, no-param-reassign
+			currentAttempt++;
+		}, intervalTime);
+	});
+const { log } = console;
+
+async function main() {
```

Comment:
Shouldn't this be making use of a test framework so that we can `expect` the results we're getting back?